lib/mongoid/extensions/date.rb

```diff
@@ -50,8 +50,9 @@ def demongoize(object)
         #
         # @param [ Object ] object The object to mongoize.
         #
-        # @return [ Time ] The object mongoized.
+        # @return [ Time | nil ] The object mongoized or nil.
         def mongoize(object)
+          return if object.nil?
```

Comment:
`return if object.blank?`. You have a redundant `unless object.blank?` on the next line.

Reply:
done